lib/charliecloud.py

```diff
@@ -341,7 +362,17 @@ def layers_read(self):
          except tarfile.TarError as x:
             FATAL("cannot open: %s: %s" % (path, x))
          members = collections.OrderedDict([(m, None) for m in members_list])
-         layers[lh] = TT(fp, members)
+         if ((lh in layers) and (len(members) > 0)):
+            FATAL("duplicate non-empty layer %s" % lh[:7])
+         if (len(members) > 0):
+            layers[lh] = TT(fp, members)
+         else:
+            empty_cnt += 1
+      if (self.schema_version == 'v1'):
+         DEBUG('reversing layer order for schema version one (1)')
+         layers = collections.OrderedDict(reversed(layers.items()))
+      if (empty_cnt > 0):
+         INFO("skipped %d empty layers" % empty_cnt)
```

Comment:
I wonder if this should be `DEBUG` and unconditional; empty layers seems like something end users might not care about and/or be confused by.